### DISCUSSION TITLE
New Data Provided by NCERT

### DISCUSSION THREAD

**Discussion by gprao08** [Sunbird-cQube/community#861] [General]
@RNarmatha 
Recently we have got NCERT new data (starter pack programs), in this regard need to we remove old data and ingest new data or directly ingest new data. Because AP School Education Department want to show latest NCERT data only in dashboard.

**Comment by RNarmatha** [comment]
Hi @gprao08 ,- Thanks for reaching us . 
@bommalla - please look into this and provided the neccessary support

**Comment by bommalla** [comment]
@gprao08 

Please find the below link and start doing the ingestion.if you have questions fell free to raise in discussions forum .

https://cqube.sunbird.org/use-cqube/ingesting-the-data/high-level-understanding-of-ingestion

**Reply by gprao08** [reply]
@bommalla 
Hi Rajesh we have issue while uploading the VSK data (latest data provided by NCERT), after National Programe ingestion no data present all dashboards shows 0%. Could you plz help me out in this issue?

**Comment by bommalla** [comment]
@gprao08 

what was the issue ? Can you please explain me in detail ?

**Reply by gprao08** [reply]
@bommalla 
After ingest latest data (removed old NCERT data) it shows all dashboards empty. Before this I have ingest State Specific data but it is also not showing in dashboard.

**Reply by gprao08** [reply]
@bommalla 
Is there any need to remove old data and go with new data? Now NCERT provided new data.

**Comment by bommalla** [comment]
@gprao08 

which version of cQube are you installed ? If you need to remove the old data from the database. please run the below command in nifi container and re ingest the data again.

sudo docker exec -it nifi_app bash
cd Sunbird-cQube-processing-ms/impl/c-qube/ingest
yarn cli nuke-db

Note: If have ingestion issue please go through the below microsite.

https://cqube.sunbird.org/deployment-process/sdc-deployment

**Reply by gprao08** [reply]
@bommalla 
Ok. We will do the process above and re ingest. Later I will post the update. We are using 5.0.3 version application. Tq.

**Reply by gprao08** [reply]
@bommalla 
Could you plz clarify the below.
We have received data from NCERT the file names differ with the actual files which you stated in Nisththa. Both the files are not same. Shall ask the data as stated in Nisththa and shall I go with the same files. Screen shots attached for your reference,
![file recommended by dev](https://github.com/Sunbird-cQube/community/assets/143190553/c8a7c472-1bc2-4bad-803a-05f252e6a339)
![files sent by ncert](https://github.com/Sunbird-cQube/community/assets/143190553/3922c5ec-c152-468f-9a48-85e5b7f5404c)

**Comment by bommalla** [comment]
@gprao08 
Please find the new cQube vsk template and check the New_nishtha tab.based on the template you can check the data which you received from the NCERT.

https://docs.google.com/spreadsheets/d/10yWmivrqtxdRLiJpX_ytzxbvoGG0WNWEI-7RLV8zelM/edit#gid=208908870

**Reply by gprao08** [reply]
@bommalla 
Thanks for sharing new template. 
As per new template I have not found the below files from NISHTHA.
1. NISHTHA_course-wise_certification
2. NISHTHA_participants_per_program
Shall I go without these files or shall I ask them to send?

**Reply by bommalla** [reply]
@gprao08 

As per the new_template we need all nishtha files then only data got ingested other wise data not ingested. Please ask them to get it.

**Reply by gprao08** [reply]
@bommalla 
ok. Tq.

**Reply by gprao08** [reply]
@bommalla 
We got the missing files as mentioned above. Now I identified filename mismatch in latest shared template above with postman collection in VSK_Grammer. Those file names in the above mentioned latest template are
NISHTHA_program_implementation_status.zip, 
NISHTHA_course-wise_certification.zip, 
NISHTHA_medium_of_instructions.zip, 
NISHTHA_Program-wise_achieved_certificates.zip, 
NISHTHA_participants_per_program.zip 

**but in the postman collection VSK_Grammer are**

diksha_nishtha_consumption-by-course
diksha_nishtha_consumption-by-district
diksha_nishtha_percentage-enrollment-certification
diksha_nishtha_program-started
diksha_nishtha_tot-courses-medium

Can I proceed with the above or shall I need to update my postman collection? If update required could you plz provide the latest postman collection?

**Reply by gprao08** [reply]
![Uploading vsk_grammer.jpg…]()

**Reply by bommalla** [reply]
@gprao08 

Please find the latest Postman collection below. Kindly import the latest Postman collections into Postman and begin the ingestion process.

[Postman_collections.zip](https://github.com/Sunbird-cQube/community/files/14329986/Postman_collections.zip)

**Reply by gprao08** [reply]
@bommalla 
After import all postman collection successfully I upload VSK_Grammer and dimension and checked in the database found 5 & 15 rows selected. Later I tried to upload through National Programs API it is showing  the below error: 
![image](https://github.com/Sunbird-cQube/community/assets/143190553/5b9ba6a7-6316-4a69-a570-3a382c642bd1)

**Comment by bommalla** [comment]
@gprao08 

API endpoint is wrong.please use below endpoint 

https://vsk.ap.gov.in/api/ingestion/national_programs

**Reply by gprao08** [reply]
@bommalla 
Thanks. Now 13 files only found in Miniobucket todays date folder. Two files those are 
1. NISHTHA_course-wise_certification.zip and 
2. vsk_diksha_etb_coverage-status.zip 
                              not found even after getting "File is being processed" message.
![image](https://github.com/Sunbird-cQube/community/assets/143190553/8f39b2fe-9178-4bd7-b9ed-28ac83787e56)

**Comment by bommalla** [comment]
@gprao08 
Thank you for the update. Could you please confirm whether the columns, data, and schema of the two files mentioned above align ?

**Reply by gprao08** [reply]
@bommalla 
Found the two files in error folder.
 
[vsk_diksha_etb_qr-coverage_errors.csv](https://github.com/Sunbird-cQube/community/files/14340402/vsk_diksha_etb_qr-coverage_errors.csv)
[NISHTHA_course-wise_certification_errors.csv](https://github.com/Sunbird-cQube/community/files/14340403/NISHTHA_course-wise_certification_errors.csv)

**Reply by gprao08** [reply]
@bommalla 
In the latest template not mentioned state_name & state_code. But is showing error must have the both coloumns.
![image](https://github.com/Sunbird-cQube/community/assets/143190553/2e6077a5-6b74-4586-90c9-4f4a90a5fcb9)

**Comment by bommalla** [comment]
@gprao08 

Kindly review both error files for any presence of null values in the data. Correct any identified null values and proceed with re-ingesting the data.

**Reply by bommalla** [reply]
@gprao08 

Please add the state_id and state_code columns in NISHTHA_course-wise_certification.csv file and re ingest the data again.

**Reply by gprao08** [reply]
@bommalla 
Done. Tq. Now all files found in Miniobucket.

**Comment by bommalla** [comment]
@gprao08 

Thanks for the update

**Reply by gprao08** [reply]
@bommalla 
After successfully completes "Run_adapters" and "data_moving_local" all dashboards shows 0 counts. Last time also we face the same issue, Could you plz help me to resolve the issue?
Moreover Even I click on any dashboard it will logout, same issue we faced earlier.

**Comment by bommalla** [comment]
@gprao08 

which version of cQube are you installed ?

**Reply by gprao08** [reply]
@bommalla 
5.0.3

**Reply by bommalla** [reply]
@gprao08 

cQube v5.0.3 version there is no "Data_moving_local" processor group.instead of data moving local please schedule "Run Latest Code local"  processor group

**Comment by bommalla** [comment]
@gprao08 

Did you schedule the ingest_data processor group  ?

**Reply by gprao08** [reply]
@bommalla 
Yes. completed
![image](https://github.com/Sunbird-cQube/community/assets/143190553/fb610553-86ae-463d-8343-19d913b86fc4)

**Comment by gprao08** [comment]
@bommalla @RNarmatha 
Thanks for your support, Now all Dashboards showing its counts except PM-Poshan. I will cross check the data and update.
